sycl/source/detail/windows_pi.cpp

```diff
@@ -66,6 +68,40 @@ void *getOsLibraryFuncAddress(void *Library, const std::string &FunctionName) {
       GetProcAddress((HMODULE)Library, FunctionName.c_str()));
 }
 
+static std::filesystem::path getCurrentDSODirPath() {
+  wchar_t Path[MAX_PATH];
+  auto Handle =
+      getOSModuleHandle(reinterpret_cast<void *>(&getCurrentDSODirPath));
+  DWORD Ret = GetModuleFileName(
+      reinterpret_cast<HMODULE>(ExeModuleHandle == Handle ? 0 : Handle),
+      reinterpret_cast<LPWSTR>(&Path), sizeof(Path));
+  assert(Ret < sizeof(Path) && "Path is longer than PATH_MAX?");
+  assert(Ret > 0 && "GetModuleFileName failed");
+  (void)Ret;
+
+  BOOL RetCode = PathRemoveFileSpec(reinterpret_cast<LPWSTR>(&Path));
+  assert(RetCode && "PathRemoveFileSpec failed");
+  (void)RetCode;
+
+  return std::filesystem::path(Path);
+}
+
+// Load plugins corresponding to provided list of plugin names.
+std::vector<std::tuple<std::string, backend, void *>>
+loadPlugins(const std::vector<std::pair<std::string, backend>> &&PluginNames) {
+  std::vector<std::tuple<std::string, backend, void *>> LoadedPlugins;
+  const std::filesystem::path LibSYCLDir = getCurrentDSODirPath();
+
+  for (auto &PluginName : PluginNames) {
+    void *Library = getPreloadedPlugin(LibSYCLDir / PluginName.first);
+    LoadedPlugins.push_back(std::make_tuple(std::move(PluginName.first),
+                                            std::move(PluginName.second),
+                                            std::move(Library)));
```

Comment:
```suggestion
                                            Library));
```

Moving a pointer is overcomplicated a little bit :)

Reply:
Fixed:)